### PR TITLE
Feature/fix chart report tables

### PR DESCRIPTION
**Pull request by xurxodev**
### :pushpin: References

-   **Issue:** Closes https://app.clickup.com/t/dz3f9q

### :memo: Implementation

- [x] Fix show chart and report tables in the object table in 2.34
- [x] Fix bug in include exclude step for charts and report tables in 2.34
- [x] Fix bug in include summary step for charts and report tables in 2.34
- [x] Fix bug to sync chart and report tables in 2.34

### :bookmark_tabs: Others

-   Any change in the [GUI library](https://github.com/EyeSeeTea/d2-ui-components)? If so, what branch/PR?

-   Any change in the [D2 Api](https://github.com/EyeSeeTea/d2-api)? If so, what branch/PR?

### CODE REVIEW

**Comment by EyeSeeTeaBotTest** [comment]
<h3>Tests failed with errors :rotating_light:</h3><br />Build commit Merge c2d2380ca6bf2a90c3de00eccd80e76638196b43 into 16c9d4668b15e21de109e775392687d053454c70 with commit id 0294a834859fd0fc487a889162a8f99981375e4a on branch feature/fix_chart_report_tables<br /><br />Travis log available at <a href=https://travis-ci.org/EyeSeeTea/metadata-synchronization/builds/759307472 target=_blank>builds/759307472</a><br /><br />See cypress results at <a href=https://eyeseetea.github.io/feedback-ci/EyeSeeTea/metadata-synchronization/pr/827/0294a834859fd0fc487a889162a8f99981375e4a target=_blank>feedback-ci/EyeSeeTea/metadata-synchronization/pr/827/0294a834859fd0fc487a889162a8f99981375e4a</a>

**Comment by adrianq** [comment]
@xurxodev This looks good to me but let's discuss tomorrow about include/exclude rules for this kind of objects.

File: src/domain/metadata/usecases/MetadataSyncUseCase.ts [hunk 1]
```diff
@@ -95,7 +95,7 @@ export class MetadataSyncUseCase extends GenericSyncUseCase {
         const metadataRepository = await this.getMetadataRepository();
         const filterRulesIds = await metadataRepository.getByFilterRules(filterRules);
         const allMetadataIds = _.union(metadataIds, filterRulesIds);
-        const metadata = await metadataRepository.getMetadataByIds<Ref>(allMetadataIds, "id");
+        const metadata = await metadataRepository.getMetadataByIds<Ref>(allMetadataIds, "id,type"); //type is required to transform visualizations to charts and report tables
```
Comment (SferaDev) [inline src/domain/metadata/usecases/MetadataSyncUseCase.ts:98]: Not sure why was this needed for

File: src/domain/synchronization/usecases/GenericSyncUseCase.ts [hunk 2]
```diff
@@ -37,7 +37,7 @@ export type SynchronizationClass =
 
 export abstract class GenericSyncUseCase {
     public abstract readonly type: SynchronizationType;
-    public readonly fields: string = "id,name";
+    public readonly fields: string = "id,name,type"; //type is required to transform visualizations to charts and report tables
```
Comment (SferaDev) [inline src/domain/synchronization/usecases/GenericSyncUseCase.ts:40]: extractMetadata

File: src/presentation/react/core/components/sync-wizard/common/SummaryStep.tsx [hunk 3]
```diff
@@ -199,7 +204,11 @@ const SaveStep = ({ syncRule, onCancel }: SyncWizardStepProps) => {
                     >
                         <ul>
                             {syncRule.excludedIds.map(id => {
-                                const element = _(metadata).values().flatten().find({ id });
+                                const values = Object.keys(metadata).map(
+                                    key => metadata[key as keyof MetadataEntities]
+                                );
+
+                                const element = values.flat().find(element => element?.id === id);
```
Comment (SferaDev) [inline src/presentation/react/core/components/sync-wizard/common/SummaryStep.tsx:211]: This breaks in old browsers (old meaning some for example mobile are this month old)

File: src/presentation/react/core/components/sync-wizard/common/SummaryStep.tsx [hunk 4]
```diff
@@ -96,7 +97,8 @@ const SaveStep = ({ syncRule, onCancel }: SyncWizardStepProps) => {
             ...syncRule.dataSyncAttributeCategoryOptions,
             ...cleanOrgUnitPaths(syncRule.dataSyncOrgUnitPaths),
         ];
-        getMetadata(api, ids, "id,name").then(updateMetadata);
+
+        compositionRoot.metadata.getByIds(ids, "id,name,type").then(updateMetadata); //type is required to transform visualizations to charts and report tables
```
Comment (SferaDev) [inline src/presentation/react/core/components/sync-wizard/common/SummaryStep.tsx:101]: Not sure why was this needed for

File: src/presentation/react/core/components/sync-wizard/metadata/MetadataIncludeExcludeStep.tsx [hunk 5]
```diff
@@ -33,24 +35,28 @@ const MetadataIncludeExcludeStep: React.FC<SyncWizardStepProps> = ({ syncRule, o
...
     const [models, setModels] = useState<typeof D2Model[]>([]);
     const [selectedType, setSelectedType] = useState<string>("");
+    const { compositionRoot } = useAppContext();
 
     useEffect(() => {
-        getMetadata(api, syncRule.metadataIds, "id,name").then((metadata: MetadataPackage) => {
-            const models = _.keys(metadata).map((type: string) => {
-                return modelFactory(type);
+        compositionRoot.metadata
+            .getByIds(syncRule.metadataIds, "id,name, type") //type is required to transform visualizations to charts and report tables
```
Comment (SferaDev) [inline src/presentation/react/core/components/sync-wizard/metadata/MetadataIncludeExcludeStep.tsx:42]: Not sure why was this needed for
